File: eastwastes/Dobbin_Crossaxe.pl
# items: 30162, 1047
my $ring_seven_final = undef;

sub EVENT_SAY { 
  if($text=~/Hail/i) {
    quest::say("We were too late but you managed to fend themk off yourselves! The Dain will be very pleased!");
    quest::say("Please friend, show me your Mithril ring and I will show you our gratitude.");
  }
}

sub EVENT_ITEM { 
  if (plugin::check_handin(\%itemcount, 30162 => 1)){
    quest::say("Thank you. $name. your deeds will be mentioned to  the Dain. Please take this note from Corbin to Garadain. It may help him to achieve victory over the enemy.");
    quest::summonfixeditem(1047); # Item: Note from Corbin
    quest::faction(406,5); # Faction: Coldain
    quest::faction(405,5); # Faction: Dain Frostreaver IV
    quest::faction(419,-10); # Faction: Kromrif
    quest::faction(448,-10); # Faction: Kromzek
    $ring_seven_final=undef;
  }

  plugin::return_items(\%itemcount);
}

#END of FILE Zone:eastwastes  ID:116129 -- Dobbin_Crossaxe

File: eastwastes/Dobbin_Crossaxe.lua
local ring_seven_ready = false;

function event_signal(e)
	if e.signal == 1 then
		ring_seven_ready = true;
		e.self:Say("We thought it was too late, the Dain will be very pleased!");
		e.self:Say("Please friend, show me your Mithril ring and I will show you our gratitude.");
	elseif e.signal == 2 then
		ring_seven_ready = false;
	end

end

function event_trade(e)
	local item_lib = require('items');
	if item_lib.check_turn_in(e.trade, {item1 = 30162}) then -- Item: Mithril Coldain Insignia Ring
		if ring_seven_ready then
			e.self:Say("Thank you. " .. e.other:GetName().. ", your deeds will be mentioned to the Dain. Please take this note from Corbin to Garadain. it may help him to achieve victory over the enemy.");
			ring_seven_ready = false;
			e.other:QuestReward(e.self,0,0,0,0,1047,10000); -- Item: Note from Corbin
			e.other:Faction(406, 5);	-- Faction: Coldain
			e.other:Faction(405, 5);	-- Faction: Dain
			e.other:Faction(419, -10);	-- Faction: Kromrif
			e.other:Faction(448, -10);	-- Faction: Kromzek
			eq.depop_all(116119);		-- NPC: Corbin Blackwell
		else
			e.self:Say(string.format("I have no need for this item %s, you can have it back.", e.other:GetCleanName()));
			e.other:SummonItem(30162); -- Item: Mithril Coldain Insignia Ring
		end
	end
	item_lib.return_items(e.self, e.other, e.trade);
end
